integration/command.go

```diff
@@ -82,6 +82,17 @@ var waitUntilInspektorGadgetPodsDeployed *command = &command{
 	done`,
 }
 
+var deploySPO *command = &command{
+	name: "Deploy Security Profiles Operator (SPO)",
+	cmd: `
+	kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.1/cert-manager.yaml
+	kubectl --namespace cert-manager wait --for condition=ready pod -l app.kubernetes.io/instance=cert-manager
+	kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/security-profiles-operator/main/deploy/operator.yaml
+	kubectl --namespace security-profiles-operator wait --for condition=ready pod -l app=security-profiles-operator
```

Comment:
Commands from:
https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/installation-usage.md#install-operator

But the docs mention that the commands for OpenShift are different. So we should detect if we're running on OpenShift here...
cc @blanquicet

Reply:
It shouldn't be a problem. I am already doing it to avoid running some gadgets that are not supported because of the kernel version. See [this draft PR](https://github.com/kinvolk/inspektor-gadget/pull/561):

https://github.com/kinvolk/inspektor-gadget/blob/2fb7a16675709e25d64c69bcc648f84c7519e92c/integration/integration_test.go#L656-L658

Reply:
I would like to adapt the commands to support OpenShift too. Besides, I added the commands in the init section, and not in a specific test. Do you think I should move them to this new test only, so it could be easily disabled based on k8sDistro?

Reply:
> Do you think I should move them to this new test only, so it could be easily disabled based on k8sDistro?
>
It is not necessary because the k8sDistro would be available also in `TestMain()`. Maybe you could do something similar to what I did to have a different initial delay based on the k8sDistro, check [here](https://github.com/kinvolk/inspektor-gadget/blob/abe3b5f049a14bc04d4ff9b9aee40aba6c7f38ee/integration/integration_test.go#L117-L130).

However, I was also wondering if maybe we should deploy the SPO only for the tests that needed it but I noticed it could take a long time to be ready because of the [webhook](https://github.com/kinvolk/inspektor-gadget/blob/d6dfbd5b31ea44a5862c65f3de5121508809c00c/integration/command.go#L87-L93) so I am not sure what is more convenient here. Maybe I would keep it in the init section for simplicity.

On the other hand, I remember I discussed with Marga about the issues of running SPO on the GitHub-hosted runners, because of the limited resources, and she suggested reducing the number of deployment replicas. They are three by default and, for instance, minikube is a single node cluster so maybe they are not all needed. What do you think? Could it significantly help to reduce the required resources so that minikube does not go OOM as Francis saw?

Reply:
I had problems with the limited resources in the GitHub-hosted runner too. I just changed the replicas and resources fields, and it works now.

wrt to deploying SPO in TestMain() or in the individual test, I am fine either way. I can keep it in TestMain() and if your PR about ARO gets merged first, I can rebase and add the `k8sDistro` test if that does not work.

cc @eiffel-fl

Reply:
> What do you think? Could it significantly help to reduce the required resources so that minikube does not go OOM as Francis saw?

If I remember correctly, I had this problem when I ran `minikube` with 2GB.
The runner has 7GB, so I think (hope?) it should suffice to deploy the SPO.